libraries/chain/include/eosio/chain/wasm_eosio_injection.hpp

```diff
@@ -789,10 +795,11 @@ namespace eosio { namespace chain { namespace wasm_injections {
          void inject() {
             _module_injectors.inject( *_module );
             // inject checktime first
-            injector_utils::add_import<ResultType::none>( *_module, u8"checktime", checktime_injection::chktm_idx );
+            if constexpr (full_injection)
+               injector_utils::add_import<ResultType::none>( *_module, u8"checktime", checktime_injection::chktm_idx );
```

Comment:
Why does checktime need to be the first import?  It seems like a pretty arbitrary requirement imposed by checktime_injection, which is inconsistent with the other injected intrinsics.

Reply:
@larryk85

Reply:
It was made that way so that we could ensure that we knew exactly which function was checktime without having to parse much by external tools, i.e. batch dump wasms and see how many checktimes where called (it is always 0).  This was also done to allow for checktimes to be added around other injected functions without having to keep any real state of the checktime function.  None the less, it doesn't have to do it that way, it was just a bit of a micro optimization at the time.